examples/hello.cpp

```diff
@@ -0,0 +1,79 @@
+#include <cstdio>
+
+#include <js/Initialization.h>
+#include <jsapi.h>
+
+/* This example illustrates the bare minimum you need to do to execute a
+ * JavaScript program using embedded SpiderMonkey. It does no error handling and
+ * simply exits if something goes wrong.
+ *
+ * To use the interpreter you need to create a context and a global object, and
+ * do some setup on both of these. You also need to enter a "request" (lock on
+ * the interpreter) and a "compartment" (environment within one global object)
+ * before you can execute code. */
+
+static JSClassOps global_ops = { nullptr, // addProperty
+                                 nullptr, // deleteProperty
+                                 nullptr, // enumerate
+                                 nullptr, // newEnumerate
+                                 nullptr, // resolve
+                                 nullptr, // mayResolve
+                                 nullptr, // finalize
+                                 nullptr, // call
+                                 nullptr, // hasInstance
+                                 nullptr, // construct
+                                 JS_GlobalObjectTraceHook };
+
+/* The class of the global object. */
+static JSClass global_class = { "HelloWorldGlobal",
+                                JSCLASS_GLOBAL_FLAGS,
+                                &global_ops };
+
+int
+main(int argc, const char* argv[])
+{
+  if (!JS_Init())
+    return 1;
+
+  JSContext* cx = JS_NewContext(8L * 1024 * 1024);
+  if (!cx)
+    return 1;
+  if (!JS::InitSelfHostedCode(cx))
+    return 1;
+
+  // Scope for our various stack objects (JSAutoRequest, JS::RootedObject), so
+  // they all go out of scope before we call JS_DestroyContext().
+  {
+    JSAutoRequest ar(cx);
+    JS::CompartmentOptions options;
+    JS::RootedObject global(
+      cx,
+      JS_NewGlobalObject(
+        cx, &global_class, nullptr, JS::FireOnNewGlobalHook, options));
+    if (!global)
+      return 1;
+
+    JS::RootedValue rval(cx);
+
+    // Scope for JSAutoCompartment
+    {
+      JSAutoCompartment ac(cx, global);
+      if (!JS_InitStandardClasses(cx, global))
```

Comment:
// Add standard JavaScript classes to the global so we have a useful environment.